src/LoginForm.jsx

```diff
@@ -16,35 +15,35 @@ import { useLogin } from "./hooks/useLogin";
 // * Show an alert box (native Javascript alert) if login succeeds. CHECK THE "login()" FUNCTION TO FIND OUT HOW TO LOGIN SUCCESSFULLY.
 
 const LoginForm = () => {
-  const { onChange, email, password, onLogin, error } = useLogin();
+  const { actions, values } = useLogin();
   return (
     <form>
       <div>
         <label>Email:</label>
         <input
           type="email"
           name="email"
-          value={email}
+          value={values.email}
           required
-          onChange={onChange}
+          onChange={actions.onChange}
         />
       </div>
       <div>
         <label>Password:</label>
         <input
           type="password"
           name="password"
-          value={password}
-          onChange={onChange}
+          value={values.password}
+          onChange={actions.onChange}
           required
         />
       </div>
       {/* Display form error messages inside the "div". Show "div" ONLY if there are login error */}
-      {!!error && (
-        <div style={{ color: "red", marginBottom: "16px" }}>{error}</div>
+      {!!values.error && (
+        <div style={{ color: "red", marginBottom: "16px" }}>{values.error}</div>
       )}
-      <button type="submit" onClick={onLogin}>
-        Login
+      <button type="submit" onClick={actions.onLogin} disabled={values.loading}>
```

Comment:
Have you noticed that the button type is "submit"? Have you ever used or tried to submit the form (onSubmit)?

Reply:
Oh, sorry, got it